idaes/surrogate/alamopy_new.py

```diff
@@ -0,0 +1,1069 @@
+            input_min.append(b[0])
+            input_max.append(b[1])
+
+        # Get number of data points to build alm file
+        n_rdata, n_inputs = training_data.shape
+
+        if validation_data is not None:
+            n_vdata, n_inputs = validation_data.shape
+        else:
+            n_vdata = 0
+
+        stream.write("# IDAES Alamopy input file\n")
+        stream.write(f"NINPUTS {len(self._input_labels)}\n")
+        stream.write(f"NOUTPUTS {len(self._output_labels)}\n")
+        stream.write(f"XLABELS {' '.join(map(str, self._input_labels))}\n")
+        stream.write(f"ZLABELS {' '.join(map(str, self._output_labels))}\n")
+        stream.write(f"XMIN {' '.join(map(str, input_min))}\n")
+        stream.write(f"XMAX {' '.join(map(str, input_max))}\n")
+        stream.write(f"NDATA {n_rdata}\n")
+        if validation_data is not None:
+            stream.write(f"NVALDATA {n_vdata}\n")
+        stream.write("\n")
+
+        # Other options for config
+        # Can be bool, list of floats, None, Enum, float
+        # Special cases
+        if self.config.monomialpower is not None:
+            stream.write(f"MONO {len(self.config.monomialpower)}\n")
+        if self.config.multi2power is not None:
+            stream.write(f"MULTI2 {len(self.config.multi2power)}\n")
+        if self.config.multi3power is not None:
+            stream.write(f"MULTI3 {len(self.config.multi3power)}\n")
+        if self.config.ratiopower is not None:
+            stream.write(f"RATIOS {len(self.config.ratiopower)}\n")
+        if self.config.custom_basis_functions is not None:
+            stream.write(
+                f"NCUSTOMBAS {len(self.config.custom_basis_functions)}\n")
+
+        for o in supported_options:
+            if self.config[o] is None:
+                # Write nothing to alm file
+                continue
+            elif isinstance(self.config[o], Enum):
+                # Need to write value of Enum
+                stream.write(f"{o} {self.config[o].value}\n")
+            elif isinstance(self.config[o], bool):
+                # Cast bool to int
+                stream.write(f"{o} {int(self.config[o])}\n")
+            elif isinstance(self.config[o], (str, float, int)):
+                # Write value to file
+                stream.write(f"{o} {self.config[o]}\n")
+            else:
+                # Assume the argument is a list
+                stream.write(f"{o} {' '.join(map(str, self.config[o]))}\n")
+
+        stream.write("\nTRACE 1\n")
+        if trace_fname is not None:
+            stream.write(f"TRACEFNAME {trace_fname}\n")
+
+        stream.write("\nBEGIN_DATA\n")
+        # Columns will be writen in order in input and output lists
+        training_data.to_string(
+            buf=stream,
+            columns=self._input_labels + self._output_labels,
+            header=False,
+            index=False,
+            float_format=lambda x: str(x).format(":g"))

```

Comment:
@andrewlee94 re: the tests failing on Python 3.6 because of extra spaces, I was thinking that it's maybe worth a quick try if something like this would address those (also applied to other places where `DataFrame.to_string()` is used:

```suggestion
            float_format=lambda x: str(x).format(":g").strip())

```